mycroft/skills/mycroft_skill/mycroft_skill.py

```diff
@@ -177,35 +187,41 @@ def dialog_renderer(self):
         # Fall back to main language
         return self.dialog_renderers.get(self._core_lang)
 
+    @property
+    def settings_path(self):
+        is_xdg = is_using_xdg()
+        if self.settings_write_path:
+            if is_xdg:
+                LOG.warning("self.settings_write_path has been deprecated!\n"
+                            "XDG standards are not assured and causes confusion regarding settings location\n"
+                            "Support for overriding settings.json path will be removed in version 0.0.3")
+            return join(self.settings_write_path, 'settings.json')
+        if not is_xdg:
+            return join(self.root_dir, 'settings.json')
```

Comment:
should this use `{data dir}/{skill_id}` instead? the old path in practice is `/opt/mycroft/skills/skill_id/settings.json`, since ovos-core supports multiple skill folder this convention is broken if we use self.root_dir, self.root_dir is no longer equivalent to what used to correspond to paths set in mycroft.conf